### PR TITLE
Upgrade to Rails 5.1.4

### PR DESCRIPTION
**Pull request by danielburnley**
Main changes required revolved around the changes to how params are
passed within tests when calling get. Rather than passing key/values
when calling get, you now need to pass them within a params hash.

In addition to this, the params hash within controllers is now an
object which has to explictly allow parameters.

Finally, we upgraded anything that was required to be upgraded to be
compatible with Rails 5

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -4,26 +4,27 @@ gem 'addressable'
...
 gem 'govuk_navigation_helpers', '~> 6.3'
 gem 'htmlentities', '~> 4.3.0'
 gem 'invalid_utf8_rejector', '~> 0.0.0'
 gem 'logstasher', '~> 1.1.0'
 gem 'plek', '~> 1.12.0'
-gem 'rack_strip_client_ip', '~> 0.0.0'
-gem 'rails', '4.2.10' # version 5 is available
-gem 'rails-i18n', '~> 4.0.0'
+gem 'rack_strip_client_ip', '~> 0.0.2'
+gem 'rails', '5.1.4' # version 5 is available
```
Comment (h-lame) [inline Gemfile]: We probably don't need the `# version 5 is available` comment any more.

File: config/initializers/cookies_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Be sure to restart your server when you modify this file.
+
+# Specify a serializer for the signed and encrypted cookie jars.
+# Valid options are :json, :marshal, and :hybrid.
+Rails.application.config.action_dispatch.cookies_serializer = :marshal
```
Comment (RoryMacDonald) [inline config/initializers/cookies_serializer.rb]: This should be :hybrid

File: config/spring.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+%w(
+  .ruby-version
+  .rbenv-vars
+  tmp/restart.txt
+  tmp/caching-dev.txt
```
Comment (RoryMacDonald) [inline config/spring.rb]: I don't think we need the spring config (not seeing it used)

File: config/puma.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+# option, you will want to use this block to reconnect to any threads
+# or connections that may have been created at application boot, as Ruby
+# cannot share connections between processes.
+#
+# on_worker_boot do
+#   ActiveRecord::Base.establish_connection if defined?(ActiveRecord)
+# end
+#
+
+# Allow puma to be restarted by `rails restart` command.
```
Comment (RoryMacDonald) [inline config/puma.rb]: We've been removing the puma configs, as unicorn seems to be running.

File: config/cable.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+development:
```
Comment (RoryMacDonald) [inline config/cable.yml]: I don't think we need actioncable - can remove this file

**Pull request merged**